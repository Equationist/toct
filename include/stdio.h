/* stdio.h - Standard Input/Output Library for CompilerKit C Compiler */

#ifndef _STDIO_H
#define _STDIO_H

/* Standard I/O types */
typedef struct {
    int _fd;
    int _flags;
    char* _buffer;
} FILE;

/* Standard streams */
extern FILE* stdin;
extern FILE* stdout; 
extern FILE* stderr;

/* Standard I/O functions */
int printf(const char* format, ...);
int fprintf(FILE* stream, const char* format, ...);
int sprintf(char* str, const char* format, ...);
int snprintf(char* str, int size, const char* format, ...);

int scanf(const char* format, ...);
int fscanf(FILE* stream, const char* format, ...);
int sscanf(const char* str, const char* format, ...);

int putchar(int c);
int puts(const char* s);
int fputs(const char* s, FILE* stream);
int fputc(int c, FILE* stream);

int getchar(void);
char* gets(char* s);
char* fgets(char* s, int size, FILE* stream);
int fgetc(FILE* stream);
int ungetc(int c, FILE* stream);

FILE* fopen(const char* filename, const char* mode);
FILE* freopen(const char* filename, const char* mode, FILE* stream);
int fclose(FILE* stream);
int fflush(FILE* stream);

long ftell(FILE* stream);
int fseek(FILE* stream, long offset, int whence);
void rewind(FILE* stream);
int feof(FILE* stream);
int ferror(FILE* stream);
void clearerr(FILE* stream);

/* File position constants */
#define SEEK_SET 0
#define SEEK_CUR 1
#define SEEK_END 2

/* End of file constant */
#define EOF (-1)

/* Buffer size */
#define BUFSIZ 8192

/* Null pointer */
#ifndef NULL
#define NULL ((void*)0)
#endif

#endif /* _STDIO_H */