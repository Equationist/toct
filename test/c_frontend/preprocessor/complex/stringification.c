/* Test stringification operator (#) */
#define STRINGIFY(x) #x
#define PRINT_VAR(var) printf(#var " = %d\n", var)
#define ASSERT_EQ(a, b) printf("Assert: " #a " == " #b " -> %s\n", (a) == (b) ? "PASS" : "FAIL")

#include <stdio.h>

int main() {
    int value = 42;
    int result = 100;
    
    printf("Stringification tests:\n");
    printf("STRINGIFY(hello) = %s\n", STRINGIFY(hello));
    printf("STRINGIFY(123) = %s\n", STRINGIFY(123));
    printf("STRINGIFY(value + 10) = %s\n", STRINGIFY(value + 10));
    
    printf("\nVariable printing:\n");
    PRINT_VAR(value);
    PRINT_VAR(result);
    
    printf("\nAssertion tests:\n");
    ASSERT_EQ(value, 42);
    ASSERT_EQ(result, 100);
    ASSERT_EQ(value * 2, 84);
    
    return 0;
}