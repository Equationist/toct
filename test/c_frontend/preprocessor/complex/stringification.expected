Stringification tests:
STRINGIFY(hello) = hello
STRINGIFY(123) = 123
STRINGIFY(value + 10) = value + 10

Variable printing:
value = 42
result = 100

Assertion tests:
Assert: value == 42 -> PASS
Assert: result == 100 -> PASS
Assert: value * 2 == 84 -> PASS